mono/mini/aot-compiler.c

```diff
@@ -10497,6 +10511,17 @@ emit_aot_file_info (MonoAotCompile *acfg, MonoAotFileInfo *info)
 	}
 }
 
+static char *
+sanitize_aot_symbol_name (char *symbol)
+{
+	char *p = symbol;
+	for (p = symbol; *p; ++p) {
+		if (!(isalnum (*p) || *p == '_'))
+			*p = '_';
+	}
+	return g_strdup (symbol);
```

Comment:
Might as well strdup it first, then replace?
Also there is probably an extra albeit harmless check for '_'.